profile/profile.js

```diff
@@ -301,13 +368,61 @@ WDN.initializePlugin('modal', [function() {
 					}
 				});
 			});
+		}
 	
+		function initializeAbilitiesEventListeners() {
+			$("#abilitiesEditButton").click(function(e) {
+				$("#abilitiesSelect").find('.editView').show();		
+				$("#abilitiesSelect").find('.preview').hide();
+				$(this).hide();
+				$("#abilitiesCancelButton").show()
+			});
+	
+			$("#abilitiesCancelButton").click(function(e) {
+				$("#abilitiesSelect").find('.editView').hide();		
+				$("#abilitiesSelect").find('.preview').show();
+				$(this).hide();
+				$("#abilitiesEditButton").show()
+			});
+	
+			$('#abilitiesSelect').on('change', function(event){
+				// abilities to be removed - all non-selected options with set ids
+				let removeUserAbilityIds = $(this).children('.editView').children('input:not(:checked)[data-userAbilityId]').map(function(index, ele){
+					return ele.dataset.userabilityid; // note this is case-sensitive and should be all lowercase
+				}).get();
+	
+				// abilities to be added - all selected options w/o ids
+				let addAbilityIds = $(this).children('.editView').children('input:checked:not([data-userAbilityId])').map(function(index, ele){
```

Comment:
I'm not confident this is doing what you think it's doing. If this is the normal javascript `.map` on an array, then the first argument is the element, and the second argument is the index (you have it switched).

You might be trying to just do:
```
blahblah.map((ele) => ele.value)
```
(same problem with the `.map` above this one)

Reply:
Wow, this has been here since literally as long as I can remember. Amazing catch.

Reply:
Ah, after doing some testing, this is actually not a JS `.map` this is a JQuery `.map` which has argument list `(index, element)`
See here: http://api.jquery.com/map/

Reply:
I wasn't sure if it was the normal `.map`, but nice job looking into it. Just gives me another reason to hate JQuery.